src/server/scripts/Spells/spell_item.cpp

```diff
@@ -3687,6 +3717,43 @@ class spell_item_taunt_flag_targeting : public SpellScript
     }
 };
 
+enum MirrensDrinkingHat
+{
+    ITEM_LOCH_MODAN_LAGER       = 23584,
+    ITEM_STOUTHAMMER_LITE       = 23585,
+    ITEM_AERIE_PEAK_PALE_ALE    = 23586
+};
+
+// 29830 - Mirren's Drinking Hat
+class spell_item_mirrens_drinking_hat : public SpellScript
+{
+    PrepareSpellScript(spell_item_mirrens_drinking_hat);
+
+    void HandleScriptEffect(SpellEffIndex effIndex)
+    {
+        uint32 itemId;
+        switch (urand(1, 6))
+        {
+            case 1:
+            case 2:
+            case 3:
+                itemId = ITEM_LOCH_MODAN_LAGER; break;
+            case 4:
+            case 5:
+                itemId = ITEM_STOUTHAMMER_LITE; break;
+            case 6:
+                itemId = ITEM_AERIE_PEAK_PALE_ALE; break;
+        }
+        if (itemId)
+            CreateItem(effIndex, itemId);
```

Comment:
There are spells to create the items, it would be nice to use them ^^
ID - 29827 Create Loch Modan Lager (SERVERSIDE)
ID - 29828 Create Stouthammer Lite (SERVERSIDE)
ID - 29829 Create Aerie Peak Pale Ale (SERVERSIDE)